php/commands/core.php

```diff
@@ -215,7 +227,20 @@ public function download( $args, $assoc_args ) {
 		WP_CLI::success( 'WordPress downloaded.' );
 	}
 
-	private static function _extract( $tarball, $dest ) {
+	private static function _extract( $tarball_or_zip, $dest ) {
+		$path_parts = pathinfo( $tarball_or_zip );
+
+		switch ( strtolower( $path_parts['extension'] ) ) {
+			case 'zip':
+				self::_extract_zip( $tarball_or_zip, $dest );
+				break;
+			default;
```

Comment:
Currently there's no validation of the extension, so I kept it has default behaviour, but I guess it doesn't hurt.